core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/OperatorUtils.java

```diff
@@ -171,4 +171,69 @@ public String toString() {
    */
   public static final BiPredicate<ExprValue, ExprValue> COMPARE_WITH_NULL_OR_MISSING =
       (left, right) -> left.isMissing() || right.isMissing() || left.isNull() || right.isNull();
+
+  /**
+   * Wildcard pattern matcher util.
+   * Percent (%) character for wildcard,
+   * Underscore (_) character for a single character match.
+   * @param pattern string pattern to match.
+   * @return if text matches pattern returns true; else return false.
+   */
+  public static boolean matches(String pattern, String text) {
+    return Pattern.compile(patternToRegex(pattern)).matcher(text).matches();
+  }
+
+  private static final char DEFAULT_ESCAPE = '\\';
+
+  private static String patternToRegex(String patternString) {
```

Comment:
1. could you seperate it from OperatorUtils in case is more focus on SQL regex to Java regex.
2. by define the ESCAPE is defined in SQL language, how do we support it?
3. is it possible to seperate the translation rule from translation logic.
for example, 
TranslationRule.Buidler.put("//%", %").put("%", ".*").build()
Then, apply rule on the input REGEX.